packages/app/src/lib/zerodev/useSmartAccount.ts

```diff
@@ -0,0 +1,586 @@
+    if (session) {
+      setSmartAccountAddress(session.smartAccountAddress);
+      setSessionExpiresAt(session.expiresAt);
+      setHasValidSession(true);
+
+      // Restore the session client
+      restoreSessionClient(session, chainId).then((client) => {
+        if (client) {
+          setSessionClient(client);
+        }
+      });
+    } else {
+      setSmartAccountAddress(null);
+      setSessionExpiresAt(null);
+      setHasValidSession(false);
+      setSessionClient(null);
+    }
+  }, [initialized, loadSession, chainId, restoreSessionClient]);
+
+  // Create a new session with ephemeral key
+  const createSession = useCallback(
+    async (
+      durationHours: number
+    ): Promise<{ success: boolean; error?: string }> => {
+      if (!walletClient || !address || !chain) {
+        return { success: false, error: 'Wallet not connected' };
+      }
+
+      if (!isSupported) {
+        return { success: false, error: 'Chain not supported by ZeroDev' };
+      }
+
+      if (!isZeroDevAvailable) {
+        return {
+          success: false,
+          error: 'ZeroDev packages not installed.',
+        };
+      }
+
+      setIsLoading(true);
+      setError(null);
+
+      try {
+        const bundlerRpc = getBundlerRpc(chainId);
+        const paymasterRpc = getPaymasterRpc(chainId);
+
+        if (!bundlerRpc) {
+          throw new Error('ZeroDev bundler RPC not configured');
+        }
+
+        const publicClient = createPublicClient({
+          chain,
+          transport: http(chain.rpcUrls.default.http[0]),
+        });
+
+        const entryPoint = getEntryPoint('0.7');
+
+        if (!KERNEL_V3_1) {
+          throw new Error('KERNEL_V3_1 constant not available.');
+        }
+
+        // Create ECDSA validator from the wallet (sudo key)
+        const ecdsaValidator = await signerToEcdsaValidator(publicClient, {
+          signer: walletClient,
+          entryPoint,
+          kernelVersion: KERNEL_V3_1,
+        });
+
+        // Generate ephemeral session key
+        const sessionPrivateKey = generatePrivateKey();
+        const sessionKeyAccount = privateKeyToAccount(sessionPrivateKey);
+
+        // Create session key signer
+        const sessionKeySigner = await toECDSASigner({
+          signer: sessionKeyAccount,
+        });
+
+        // Calculate expiry
+        const maxDurationHours = SESSION_KEY_DEFAULTS.maxDurationSeconds / 3600;
+        const clampedHours = Math.min(durationHours, maxDurationHours);
+        const expiresAt = Date.now() + clampedHours * 60 * 60 * 1000;
+
+        // Create permission validator with the session key
+        // This allows the session key to sign any call (permissive for now)
+        const permissionValidator = await toPermissionValidator(publicClient, {
+          entryPoint,
+          kernelVersion: KERNEL_V3_1,
+          signer: sessionKeySigner,
+          policies: [
+            // You can add more restrictive policies here if needed
```

Comment:
shouldn't we add some policies here?

Reply:
we should add a policy as a follow up at the moment I just turned the paymaster on for everything on zerodev's ui